### PR TITLE
WIP: Setting Stages, UI overhaul, & More Functionality

### PR DESCRIPTION
**Pull request by kavanparam**
**NOTE: this is a WIP PR. Please ignore the comments & logs. They will be removed in the coming PR.

**Halfway point PR.** Going to go ahead and merge and will make changes in the next PR.

### Changes include:
1. Welcome stage with animated entry — prompting user to read game rules & change their default name
image
2. Reactive toggle selector to show/hide output of the dice roll blocks — this block also now displays name inputted into the placeholder
image
image
3. A new dropdown menu to contain various functions to be built later — restart button was moved here & is functional
image
4. A game rules stage — presenting a modal on screen
image
5. Some responsive styling — for the GameBoard and Dice components
image
6. Animated gameplay — dice simulate a rolling movement

Pending changes:
- more conditional rendering to display the winner (if any) — which will set new Stages & output an alert modal

### CODE REVIEW

File: components/math/multiplicationConnect/GameBoard.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ const GameBoard: FC = () => {
         {grid.map((blockData) => (
           <div
             key={blockData.id}
-            className={`border-r-2 border-b-2 md:h-32 sm:h-24 h-20 border-b-[#149ECA] border-r-[#149ECA] bg-blue-900
+            className={`border-r-2 border-b-2 md:h-32 sm:h-24 h-20 border-b-[#149ECA] border-r-[#149ECA] bg-gradient-to-br from-blue-800  to-indigo-900
```
Comment (kwiedenhaupt) [inline components/math/multiplicationConnect/GameBoard.tsx:15]: nice shading

File: components/math/multiplicationConnect/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React from "react";
+
+/* todo: Modal following WDS tutorial using React portals
```
Comment (kwiedenhaupt) [inline components/math/multiplicationConnect/Modal.tsx:3]: i'm looking forward to seeing how this is implemented

File: pages/studentPortal/labs/multiplication-connect/Index.tsx [hunk 3]
```diff
@@ -22,30 +28,69 @@ const Index: FC = () => {
...
+            className="font-mono font-bold text-gray-900 bg-white border border-gray-300 focus:outline-none hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 rounded-lg px-5 py-2.5 mr-2 mb-2 dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700"
+            onClick={() => dispatch(setStage(Stage.GAME_RULES))}
+          >
+            📝 Game Rules
+          </button>
+
+          {stage === Stage.GAME_RULES && (
+            <Modal
+              type="centered"
+              closeModal={() => dispatch(setStage(Stage.GAME_PLAY))}
```
Comment (kwiedenhaupt) [inline pages/studentPortal/labs/multiplication-connect/Index.tsx:73]: very interesting how you use the modals here.. nice.

File: pages/studentPortal/labs/multiplication-connect/Index.tsx [hunk 4]
```diff
@@ -22,30 +28,69 @@ const Index: FC = () => {
...
+            >
+              <section>
+                <h2 className="text-4xl font-bold">Game Rules</h2>
+                <ol className="pt-8 space-y-2 list-decimal">
+                  <li>Roll two dice.</li>
+                  <li>Add the numbers together, and multiply by 2.</li>
+                  <li>Select this number within a block in the game.</li>
+                  <li>Repeat for the next player.</li>
+                  <li>
+                    Play until a player connect 4 squares in a row (vertically,
```
Comment (kwiedenhaupt) [inline pages/studentPortal/labs/multiplication-connect/Index.tsx:83]: just make the word plural -- play until a player "connects"

**Review by kwiedenhaupt — COMMENTED** [review]
I know it's a WIP -- very interested to see how you finish coding the modals and the changing of the stages.

File: components/math/multiplicationConnect/Modal.tsx (hunk 2 above)
Comment (kavanparam) [reply components/math/multiplicationConnect/Modal.tsx]: Implemented, you can check out the branch! Didn't use Portals though, just ended up using an overlay and changing the z-index of elements to make sure they appear over/under.

File: pages/studentPortal/labs/multiplication-connect/Index.tsx (hunk 4 above)
Comment (kavanparam) [reply pages/studentPortal/labs/multiplication-connect/Index.tsx]: ty!

File: components/math/multiplicationConnect/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import React, { FC, Ref, useRef } from "react";
+
+interface ModalProps {
+  type: String;
```
Comment (vithushan19) [inline components/math/multiplicationConnect/Modal.tsx:4]: instead of a string type you can do "centeted" | "fullscreen" | "alert"

**Review by vithushan19 — COMMENTED** [review]
lgtm

File: components/math/multiplicationConnect/Modal.tsx (hunk 5 above)
Comment (kavanparam) [reply components/math/multiplicationConnect/Modal.tsx:4]: fixed!

**Pull request merged**